source_control/gh_keys.py

```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2013, André Paramés <git@andreparames.com>
+# Based on the Git module by Leonardo Comelli <leonardo.comelli@gmail.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+import json
+import requests
+from ansible.module_utils.basic import *
+
+DOCUMENTATION = '''
+---
+module: gh_keys
+version_added: n/a
+short_description: Manages github ssh keys.
+description:
+    - The module manages the ssh key for a specific user through Github API v3.
+options:
+  action:
+    description: This tells the gh_keys module what you want it to do.
+    required: true
+    choices: 
+      list_keys:
+        description: lists keys for authenticated or non-authenticated user. If the passwd field is informed, the module returns all keys and all data about its (In this case, the module will send the authentication data in request). Otherwise, the module returns only some data about the ssh keys of non-authenticated user.
+        required_fields: [user | user & passwd]
+      get_key:
+        description: gets a specific key by github ssh-key-id
+        required_fields: [user & passwd & key_id]
+      add_key:
+        description: adds a new public ssh key
+        required_fields: [user & passwd & title & key]
+      remove_key:
+        description: removes a specific key by github ssh-key-id
+        required_fields: [user & passwd & key_id]
+  user:
+    description: Github username.
+    required: true
+  passwd:
```

Comment:
Is there a valid use for this being shortened?  Seems like using the full word would be beneficial.